examples/general/abs_home_path/src/lib.rs

```diff
@@ -178,52 +124,94 @@ fn ui() {
+                );
+            },
+        },
+        TestCase {
+            manifest_path: "ui_test/Cargo.toml",
+            context_name: "test context",
+            assert_fn: |output: &Output| {
+                let stderr = String::from_utf8_lossy(&output.stderr);
+                // We expect warnings for non-test functions but not for test functions
+                assert!(
+                    stderr.contains("this path might not exist in production"),
+                    "Expected the abs_home_path lint to trigger for non-test functions in test context: {stderr}"
+                );
+                // Verify the test detected non-test functions correctly
+                assert!(
+                    stderr.contains("non_test_function_with_home_path")
+                        || stderr.contains("src/main.rs:9:13"),
+                    "Expected warning in non-test function in test context: {stderr}"
+                );
+                // Verify that test functions weren't warned about
+                assert!(
+                    !stderr.contains("test_function_with_home_path")
+                        && !stderr.contains("src/main.rs:15:13"),
+                    "The abs_home_path lint should not trigger in test functions: {stderr}"
+                );
+            },
+        },
+    ];
+
+    // Skip tests if repository is not stored in the user's home directory
     if let Some(home) = home::home_dir()
         && !Path::new(env!("CARGO_MANIFEST_DIR")).starts_with(home)
     {
         #[expect(clippy::explicit_write)]
         writeln!(
             stderr(),
-            "Skipping `build_script_allowance` test as repository is not stored in the user's home directory"
+            "Skipping `context_allowance` tests as repository is not stored in the user's home directory"
         )
         .unwrap();
         return;
     }
 
-    // Use the cargo_dylint function to get the path to the cargo-dylint executable
     let cargo_dylint = dylint_internal::testing::cargo_dylint("../../..").unwrap();
 
-    // Run dylint on the test package with our lint
-    let output = Command::new(cargo_dylint)
-        .env("DYLINT_RUSTFLAGS", "--deny warnings")
-        .args([
-            "dylint",
-            "--manifest-path",
-            "ui_build_script/Cargo.toml",
-            "--path",
-            env!("CARGO_MANIFEST_DIR"),
-        ])
-        .logged_output(false)
-        .expect("Failed to execute cargo-dylint command");
-
-    let stderr = String::from_utf8_lossy(&output.stderr);
-
-    assert!(
-        output.status.success(),
-        "Cargo dylint check failed: {stderr}"
-    );
-    assert!(
-        !stderr.contains("this path might not exist in production"),
-        "The abs_home_path lint was incorrectly triggered in a build script: {stderr}"
-    );
+    for case in test_cases {
+        println!("Testing {} allowance...", case.context_name);
+        let output = Command::new(&cargo_dylint)
```

Comment:
```suggestion

        let output = Command::new(&cargo_dylint)
```
Nit (spacing again).